tensorboard/webapp/util/matcher.ts

```diff
@@ -16,15 +16,26 @@ limitations under the License.
  * @fileoverview Helps matching runs.
  */
 
-import {Run} from '../runs/types';
-
-export interface RunWithExperimentName extends Run {
+export interface RunMatchable {
+  runName: string;
   experimentName: string;
   experimentAlias: string;
 }
 
+/**
+ * Matches an entry based on regex and business logic.
+ *
+ * - An empty regex string always returns true.
+ * - Invalid regex always return false.
+ * - Regex matches are case insensitive.
+ * - Regex matches are not anchored.
+ * - Matches name of a run.
```

Comment:
Added the clarifying text. However, given the title of the jsdoc states, 

>  Matches an entry based on regex and business logic.

I think it would be self-explanatory that matches are against regex. :(